src/bracketology_utilities.py

```diff
@@ -0,0 +1,145 @@
+"""Bracketology Utilities."""
+
+import random
+
+
+class Team:
+    """Class representing a team in a tournament."""
+
+    def __init__(
+        self, name: str, seed: str, region: str, win_probs: dict[str, float]
+    ):
+        """Initialize the team with its name, seed, region, and win_probs."""
+        self.name = name
+        self.seed = seed
+        self.region = region
+        self.win_probs = (
+            win_probs  # {"Rd2": 99.5, "Swt16": 84.6, ..., "Champ": 22.9}
+        )
+
+    def get_win_prob(self, round_name: str) -> float:
+        """Get the win probability for a specific round."""
+        return self.win_probs.get(round_name, 0.0)
+
+    def __repr__(self) -> str:
+        """Return a string representation of the team."""
+        return f"{self.seed} {self.name}"
+
+    def __eq__(self, other: object) -> bool:
+        """Check equality of two teams."""
+        return (
+            isinstance(other, Team)
+            and self.name == other.name
+            and self.seed == other.seed
+            and self.region == other.region
+            and self.win_probs == other.win_probs
+        )
+
+
+class Bracket:
+    """Class representing a tournament bracket."""
+
+    def __init__(self, teams: list[Team]):
+        """Initialize the bracket with a list of teams."""
+        self.teams = sorted(
+            teams, key=lambda t: int(t.seed)
+        )  # Sort by seed numerically
+        self.rounds: list[list[Team]] = []
+
+        # Define round keys for matching win_probs dict keys
+        self.round_keys = [
+            "Rd2",
+            "Swt16",
+            "Elite8",
+            "Final4",
+            "Final",
+            "Champ",
+        ]
+
+    def simulate_round(self, teams: list[Team], round_name: str) -> list[Team]:
```

Comment:
`self.teams` is already an attribute of `Bracket`. Do you want to use that one instead of passing `teams` in here?

Reply:
Ah, I see below that the input `teams` here are just the ones playing in this round. Maybe clarify by calling it something else, like "remaining_teams"?